### ISSUE TITLE
Possible bug in no-nested-subscribe

### ISSUE THREAD

**Issue by fchiumeo** [cartant/eslint-plugin-rxjs#67]
The `no-nested-subscribe` rule should only report subscribe within a subscribe callback or within any operator callback ?
Because it's currently reporting an error in a subscribe within a map  callback
(in package rxjs-tslint-rules it's OK)

```typescript
from(/* Observable */)
    .pipe(
        map((items: any[]) =>
        {
            form.get('control').valueChanges
                //  error  Nested subscribe calls are forbidden  rxjs/no-nested-subscribe
                .subscribe((billing: boolean) => 
                {
                    // code
                });
        }))
    .subscribe(() =>
    {
        //more code
    });
```

**Comment by cartant** [comment]
Nope. The rule is working as intended. Explicit `subscribe` calls within composed observables should be avoided.

**Comment by cartant** [comment]
This should be fixed in 3.1.5. I was mistaken and it _is_ a bug, because usage like this is fine:

https://github.com/cartant/eslint-plugin-rxjs/blob/213794bcbb81bcfca8e5905c2520be2372ff58df/tests/rules/no-nested-subscribe.ts#L64-L68

However, usage in callbacks passed to operators is probably still a problem. At some stage, I might extend the rule to effect failures for explicit `subscribe` calls in functions passed to operators.
